src/SkosConceptHierarchicalHandler.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages;
+
+use Drupal\rdf_skos\Entity\Concept;
+
+/**
+ * Implments list pages hierarchy logic for skos concepts.
+ */
+class SkosConceptHierarchicalHandler implements HierarchicalHandlerInterface {
```

Comment:
Extend from `EntityHandlerBase`. You can also make your own base class extending this that then each of your handlers can extend.